### ISSUE TITLE
Activating Zu Translate breaks translation with TinyMCE

### ISSUE THREAD

**Issue by terzag** [picasso/zu-translate#2]
Hello,
I've recently replaced qTranslate-X by qTranslate-XT on a website and found about your plugin to enhance it with translations inside the block editor (Gutenberg). While this works well, it seems that it also makes qTranslate-XT to stop working with TinyMCE, which is still used by some plugins. In my case, WooCommerce.

When Zu Translate is enabled, editing a WooCommerce product page, which uses TinyMCE will show all languages in the editor (with the language tags unfiltered) and when switching language buttons, the editor shows the same content (i.e. modifying a language puts the same content for the others).

I assume it's a bug, unless there are some specific settings needed in qTranslate-XT or Zu Translate to have both Gutentberg and TinyMCE working?

Both plugins are up to date (qTranslate-XT 3.11.4 / Zu Translate 1.3.2).

**Comment by picasso** [comment]
I didn't expect `TinyMCE` to be used in conjunction with the block editor (Gutenberg) at all. `Zu Translate` simply disables `TinyMCE` support because it works differently and interferes with `Zu Translate`.
That's why TinyMCE and Gutenberg don't work together. But they do not work only inside the block editor, nothing is disabled in other places.

Why do they need to work together? If `WooCommerce` uses custom blocks, then you just need to register them in `Zu Translate`.

**Comment by terzag** [comment]
> Why do they need to work together? If `WooCommerce` uses custom blocks, then you just need to register them in `Zu Translate`.

Well, WooCommerce doesn't use Gutenberg at all. It has custom admin pages to enter data with various custom fields and meta boxes (e.g. a product page will have the price, quantity in stock and so on), and for the product description it has a TinyMCE text area.

> But they do not work only inside the block editor, nothing is disabled in other places.

That's weird: as soon as I enable Zu Translate, WC admin pages stop working properly (for the translation part). I've just noticed that if I enable translation buttons on all post/pages, they work in the WP post/pages list but not on the WC products list page (they just do nothing).

**Comment by picasso** [comment]
>  if I enable translation buttons on all post/pages, they work in the WP post/pages list but not on the WC products list page 

This is clear. I tested only with regular post/pages. I think that WC products is a __custom type__ and I have not implemented any special processing for them. In general, if WooCommerce uses its pages, then my code should not interfere. Have you tried on other pages (except WooCommerce and the Blocks Editor)? For example, on the `wp-admin/options-general.php`?

> It has custom admin pages to enter data with various custom fields and meta boxes (e.g. a product page will have the price, quantity in stock and so on), and for the product description it has a TinyMCE text area.

Probably, they use the same URL as and Gutenberg and therefore TinyMCE functionality is turned off.
Send me a URL of some WooCommerce page. From `wp-admin`, of course

**Comment by terzag** [comment]
It works fine in options-general.php. Also, it works fine with pages related to a custom post type.

The WC pages that are broken have a url like mywebsite.com/wp-admin/edit.php?post_type=product (list of products, with the language switcher not working but the post titles showing only the default language) or mywebsite.com/wp-admin/post.php?post=456&action=edit (product page, where I first noticed the issue in the TinyMCE text area showing all the languages with the special tags, though I noticed afterwards that the title field has the same issue).

But, as said just above, a page listing for a custom post type (e.g. mywebsite.com/wp-admin/wp-admin/edit.php?post_type=projet) still has a functional language switcher.

So, unlike what I first thought, the problem is likely not related to TinyMCE but having Zu Translate enabled seems to break qTranslate-XT on some pages (WooCommerce only?). I assume that the language switcher in the admin uses a JS script to filter the languages and toggle them, and that something is breaking that script on some pages.

**Comment by picasso** [comment]
I installed WooCommerce and working on the solution. In the near future, I publish it.

**Comment by terzag** [comment]
Thanks a lot.

**Comment by picasso** [comment]
Try version `1.3.3`

On the plugin settings page, in the __"Ignore Custom Post Types"__ block, make sure that the custom post type contains `product`. You can click the __Reset Block Editor Settings__ button.

**Comment by terzag** [comment]
Just did a test and it seems to work. I've found a weird case related to coupons, though, but it might be a bug in qTranslate-XT:
- If I edit a coupon code with Zu Translate base settings, the language switcher on the edit page is gone;
- If I add the shop_coupon to the list of cpt to ignore, it's back (so far so good); but:
- If I toggle the option to show the language switcher on all pages the list of coupons shows their name (title) blank.

The thing that makes me think that there is an issue with qTranslate-XT is that when disabling Zu Translate, the coupon edit page has a language switcher but the same text for the description is used for all languages (i.e. changing it and switching language has it changed too, similar to the issue for which I opened the current bug report) while the title, which is the code that should be common to every language, can be changed.

The part that seems to be a bug with Zu Translate is the last one, with enabling the language switcher on all pages makes the post title disappear on the coupons list page.

And I've just noticed another bug when fiddling with WC but it's not (really) related to it: when editing customers account from WooCommerce → Customers → clicking on one will open the user-edit.php?user_id=xxx, which is actually the user edit page from WP. When the "Add language switcher to all pages/posts" option from Zu Translate is enabled, there'll be a switcher on that user edit page that just does nothing (which is expected I guess, as it's a WP settings page).
The only thing related to WC is that if you edit your own admin account as a customer, it'll send you to the profile page but with the user-edit one (i.e. it'll be user-edit.php?user_id=1) instead of profile.php; on the profile.php page, the language switcher does not appear.

**Comment by picasso** [comment]
> I've found a weird case related to coupons

I do not really understand how WooCommerce works. What result should be in the Coupons section? Just remove `Language Switcher` from this page? As far as I understand the coupons have no title, only code.

**Comment by terzag** [comment]
The Coupons have a title and description, like regular content, and the title is used as the Coupon code. But I guess I should first report the issue to qTranslate-XT as it doesn't seem to work properly (coupons titles/codes should not be translatable).

The only issue related to Zu Translate is that adding the language switcher to the coupons page displays no title for the posts, which is a bit weird as other similar pages work fine.

**Comment by picasso** [comment]
My qTranslate-XT version does not try to translate Coupon Code. I fixed this error and the language switcher will not appear on the coupon page. But I will not publish this change and try to see the customers problem first.

**Comment by terzag** [comment]
Ok. I've opened a bug report on qTranslate-XT regarding the Coupons codes translation, as with Zu Translate disabled it doesn't seem to work properly anyway.

**Comment by picasso** [comment]
Fix in version 1.3.4

On the plugin settings page, open  __"Advanced Setting"__ panel (maybe enable it first in `Screen Options`) and then click the __Reset Advanced Settings__ button.

**Comment by terzag** [comment]
Thanks, I'll check that ASAP.

**Comment by picasso** [comment]
> In Screen options, I only have "Block editor support" (enabled), "Language switcher" (disabled)

something wrong with your plugin... "Language switcher" was removed 2 versions ago. Please check that the version of plugin is `1.3.4` or `1.3.5` (__Plugin info__ in right sidebar)

**Comment by terzag** [comment]
Not sure if I did something wrong previously: every time I disabled Zu Translate, deleted the content of the plugin's folder and replaced it with the new version, then re-enabled the plugin. The right version was displayed in the Plugins page of WP but I didn't check if the proper one was displayed on the actual plugin's setting page. Anyway, I've upgraded to 1.3.5 and it seems to work now, I have the Advanced Settings in the screen options (no more "Language switcher" instead).

Everything seems to be ok.
